MqttOutput.go

```diff
@@ -0,0 +1,70 @@
+package dnstap
+
+import (
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"google.golang.org/protobuf/proto"
+)
+
+// MqttOutput implements a dnstap Output that publishes dnstap data to an MQTT broker.
+type MqttOutput struct {
+	outputChannel chan []byte
+	client        mqtt.Client
+	baseTopic     string
+	qos           byte
+	wait          chan bool
+	log           Logger
+}
+
+// NewMqttOutput creates a MqttOutput for publishing dnstap data to an MQTT broker.
+func NewMqttOutput(opts *mqtt.ClientOptions, baseTopic string, qos byte) (o *MqttOutput, err error) {
+	o = new(MqttOutput)
+	o.baseTopic = baseTopic
+	o.qos = qos
+	o.outputChannel = make(chan []byte, outputChannelSize)
+	o.wait = make(chan bool)
+	client := mqtt.NewClient(opts)
+	if token := client.Connect(); token.Wait() && token.Error() != nil {
+		return nil, token.Error()
+	}
+	o.client = client
+	return
+}
+
+// SetLogger configures a logger for error events in the MqttOutput
+func (o *MqttOutput) SetLogger(logger Logger) {
+	o.log = logger
+}
+
+// GetOutputChannel returns the channel on which the MqttOutput accepts dnstap data.
+//
+// GetOutputChannel satisfies the dnstap Output interface.
+func (o *MqttOutput) GetOutputChannel() chan []byte {
+	return o.outputChannel
+}
+
+// RunOutputLoop receives dnstap data sent on the output channel, unmarshalls it
+// and publishes it to the mqtt broker
+//
+// RunOutputLoop satisfies the dnstap Output interface.
+func (o *MqttOutput) RunOutputLoop() {
+	dt := &Dnstap{}
+	for frame := range o.outputChannel {
+		if err := proto.Unmarshal(frame, dt); err != nil {
```

Comment:
We should write a separate .proto file, or extend the existing one so we can unmarshal only the type of the message and ignore other fields. This will lead to less allocation and garbage collection, and hopefully a faster deserialization here as well.

I've heard there are ways to extract a single field without deserializing the data, this may be another way to go.